pkg/sql/contention/txnidcache/concurrent_write_buffer.go

```diff
@@ -20,6 +20,10 @@ const messageBlockSize = 1024
 
```

Comment:
1024 feels pretty big to me. How do you feel about maybe 170? If each `ResolvedTxnID` is 24 bytes, that gives you as close as you can get with a 4k page and your nodes would be still under 4k too. I think at 170 you've already amortized a ton.